sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -29,6 +29,48 @@ export enum DispositionType {
   defer = "defer"
 }
 
+/**
+ * Checks if a message is settleable (ie, has settlement methods). If so, returns it, properly typecast so it
+ * can be called.
+ *
+ * If not, throws an error indicating that it can't be settled.
+ *
+ * @internal
+ * @ignore
+ */
+export function throwIfNotSettleableMessage(
+  message: ServiceBusReceivedMessage | ServiceBusMessageImpl
+): ServiceBusSettleableMessageImpl {
+  if (!isSettleableMessage(message)) {
+    throw new Error(
+      "Only messages from a receiver in 'peekLock' mode using receiveMessages(), subscribe() or the getMessageIterator() iterator can be settled."
+    );
+  }
+
+  return message;
+}
+
+/**
+ * Typeguard to check if a message is settleable.
+ *
+ *@internal
+ *@ignore
+ */
+export function isSettleableMessage(
+  origMessage: ServiceBusReceivedMessage | ServiceBusSettleableMessageImpl
+): origMessage is ServiceBusSettleableMessageImpl {
+  const message = origMessage as ServiceBusSettleableMessageImpl;
+
+  return !!(
+    message.lockToken && // messages received in receiveAndDelete mode (or peeked messages) will not have a lock token
+    // any message that we've created will also be missing settlement methods
```

Comment:
This will not be true after we remove the settlement methods altogether from the message. See https://github.com/Azure/azure-sdk-for-js/issues/9944

Any reason to not just rely on the absence of lock token to determine whether or not the message is settleable?

Reply:
Looks like we can also check for the absence of `delivery` on the message to determine if it was a peeked message.

Please see #12216 where I am moving the methods out of the ServiceBusMessageImpl  + throwing error when trying to settle peeked message. Based on how we decide to move forward, we can remove the checks and rely on service error if needed.

Reply:
I prefer this method, TBH. There's no reason for the message to have these methods at all and there's no reason to throw if we can just not export the methods in the first place.

Reply:
So, are you saying that we should not be removing the methods from `ServiceBusMessageImpl`?

Reply:
Yeah, I basically have two classes - one for non-settleable messages and one for settleable messages.